### ISSUE TITLE
Additional comments

### ISSUE THREAD

**Issue by fxpineau** [ivoa/voparquet#5]
Mark, thank you again for your work.

Here a few minor comments, to be picky and have something to say.
I am Ok if you don't "buy".

## Abstract and Introduction: "Efficient"

The word "efficient" is associated to Parquet in several places:
* first sentence of the abstract;
* first sentence of the second paragraph of the Introduction;
* last sentence of the third paragraph of the Introduction;
* "fast processing" in the second paragraph of the Introduction.

For case 1 and 3, I would replace "efficient/cy" by "**storage** efficient/cy".
I am quite against stating "efficiency" out of context here.
When compression is used, although Parquet is storage efficient, it is CPU intensive.
When accessing lot of the columns of a few rows, Parquet is not efficient.
When filtering a full file with a criteria selecting a large fraction of rows homogeneously distributed, Parquet is not efficient.
When making plots on a few columns locally, although Parquet is much more efficient than row-oriented formats, it is even more efficient to use an arrow dump of the data (you can memmap), provided you have enough disk space (, or to use Feather?).
...

Case 2 and 4, I would replace "fast processing" by the slightly more restrictive "fast **analytical** processing".
It excludes, e.g., full file filtering (with all columns in output) or conversion to a row-oriented format.

## Introduction: "Minimum requierement is column attribute"

What about adding, somewhere after "information relating to", "formatting such as width,  precision or null value"?

## 2.3 Unsigned types

As suggested by you last sentence, the following may be a discussion for a later version of the note.
But since I have those questions in mind, I am writing them down.

Paragraph 2: "can store [...] unsigned 64-integer".
Yes, but not only 64-bit integers.
As you known, Parquet also supports 8-bit signed integers and 16 and 32-bit unsigned integer,  see [parquet logical types](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md).
None of them are (yet?) supported in VOTable. 

As you also know, I would advocate for `unsigned` (and `signed`) `xtype` in VOTable to support those types (and also allowing to support 64-bit unsigned integers) without changing the VOTable standard itself.

But what to do in the current status if a Parquet file contains, e. g., a 32-bit unsigned integer column?
* declare the field as a VOTable `int` (same number of bytes)?
* declare the field as a VOTable `long` (to ensure that all possible unsigned values can be represented with the VOTable signed type)?
* check for all values and choose according to the `MAX` value (with `MIN`>=0 and `MAX` to be possibly added to the `FIELD`)?
* ...

For the client, it is not that important since the document mention that "Parquet datatypes [are] authoritative".
But should we say something for VOParquet writers (or since it is not important for readers, writers should be free to do what they want)?

In the document, Parquet files are annotated with VOTable metadata, but nothing is said about possible format conversion.
A broader question is: do we want to try, at term, to maximize the possibility of round-trip conversion between a VOParquet file and a VOTable (I would like so),  or is it already a non-goal?

**Comment by mbtaylor** [comment]
@fxpineau thanks for these comments.  I've made a PR #6 that I hope addresses them. Some responses below.

_Efficiency:_ advocacy is certainly not the point here, and I agree in any case with your comments about the limitations of parquet for resource optimisation.  I've reworded some of the discussion along the lines you suggest.

_Minimum requirement is column attribute:_ storing the null value is not one of the requirements here, since parquet handles NULLs in its own way without requiring magic values.  I have added "precision", but I don't want to write a comprehensive list here, the point is just to note that there are some per-column items that are generally required for common VO usage scenarios.

_Unsigned types:_ indeed the 64-bit integer type is just mentioned here as an example and there are other types for which similar considerations apply, and I do think that these questions are worth addressing, but that it should be done in a future evolution of this document.  The reason is that there is scope for disagreement about the best way to annotate these, which means (a) we may have better ideas about how to do it if we gain some implementation experience in future and (b) trying to agree on it now might delay the publication of this Note.  For instance I don't want to commit this document to a particular way of encoding unsigned 64-bit integers that might conflict with future developments of other standards e.g. possible introduction in DALI of a `uint64` xtype.  I've added a comment in the Introduction that tries to clarify this thinking (should there be a similar comment in sec 2.3 explaining why advice here is deliberately vague in cases of mismatch?)
